package NClustering.NProto;

enum EClusterWeightsType {
    Empty = 0; // for no-weights aggregators
    ScaledSum = 1;
    AvgNearestN = 2;
    AvgClusterTopN = 3;
    LogCount = 4;
}

message TClusterWeightsType {
    required EClusterWeightsType WeightType = 1;
    optional uint32 Num = 2; // for AvgNearestN and AvgClusterTopN types
}

message TKMeansParams {
    optional uint32 MaxIterations = 1 [default = 10000];
    optional uint32 NumAttempts = 2 [default = 5];
    optional double Epsilon = 3 [default = 0.0001];
}

message TClusteringParams {
    optional uint32 NumClusters = 1 [default = 1];
    repeated TClusterWeightsType WeightTypes = 2;
    oneof TypeSpecificParams {
        TKMeansParams KMeansParams = 3;
    }
}
